build.gradle.kts

```diff
@@ -156,147 +161,146 @@ subprojects {
-         * https://github.com/johnrengelman/shadow/blob/master/build.gradle
-         */
-
-        // make build task depend on shadowJar
-        val build: DefaultTask by tasks
-        val shadowJar = tasks["shadowJar"] as ShadowJar
-        build.dependsOn(shadowJar)
 
-        tasks.withType<ShadowJar> {
-            // ensure that a single jar is built which is the shadowed one
-            // Using non-deprecated archiveClassifier.set doesn't seem to work.
-            @Suppress("DEPRECATION")
-            classifier = ""
-            // no need to add analyzer.init.gradle in the jar since it will never be used from inside the plugin itself
-            exclude("analyzer-init.gradle")
-            // Minimise the resulting uber-jars to ensure we don't have massive jars
-            minimize() {
-               // Sometimes minimisation takes away too much ... ensure we keep these.
-               exclude(dependency("com.fasterxml.jackson.core:.*:.*"))
-               exclude(dependency("org.commonjava.maven.ext:.*:.*"))
-               exclude(dependency("org.commonjava.maven.atlas:.*:.*"))
-               exclude(dependency("org.aeonbits.owner:.*:.*"))
-               exclude(dependency("org.slf4j:.*:.*"))
-               exclude(dependency("org.apache.maven:.*:.*"))
-           }
-            doFirst {
-                manifest {
-                    attributes["Built-By"] = System.getProperty("user.name")
-                    attributes["Build-Timestamp"] = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ").format(Date())
-                    attributes["Scm-Revision"] = Grgit.open(mapOf("currentDir" to project.rootDir)).use { g -> g.head().id }
-                    attributes["Created-By"] = "Gradle ${gradle.gradleVersion}"
-                    attributes["Build-Jdk"] = System.getProperty("java.version") + " ; " + System.getProperty("java.vendor") + " ; " + System.getProperty("java.vm.version")
-                    attributes["Build-OS"] = System.getProperty("os.name") + " ; " + System.getProperty("os.arch") + " ; " + System.getProperty("os.version")
-                    attributes["Implementation-Version"] = "${project.version}"
-                }
+    apply(plugin = "signing")
+    apply(plugin = "maven-publish")
+    apply(plugin = "com.github.johnrengelman.shadow")
+    apply(plugin = "net.linguica.maven-settings")
+
+    /**
+     * The configuration below has been created by reading the documentation at:
+     * https://imperceptiblethoughts.com/shadow/plugins/
+     *
+     * Another great source of information is the configuration of the shadow plugin itself:
+     * https://github.com/johnrengelman/shadow/blob/master/build.gradle
+     */
+
+    // make build task depend on shadowJar
+    val build: DefaultTask by tasks
+    val shadowJar = tasks["shadowJar"] as ShadowJar
+    build.dependsOn(shadowJar)
+
+    tasks.withType<ShadowJar> {
```

Comment:
Doesn't this if structure mean that the common library will be published as a Shadow publication?